### PR TITLE
Use KAMU_CONTAINER_RUNTIME_TYPE env var in Makefile

### PR DESCRIPTION
**Pull request by demusdev**
## Motivation

Being able to use `podman` for e2e test.

## Checklist before requesting a review

- [ ] Unit and integration tests added
  
- [ ] Compatibility:
    
  - [ ] Network APIs: ✅
    
  - [ ] Workspace layout and metadata: ✅
    
  - [ ] Configuration: ✅
    
  - [ ] Container images: ✅
- [ ] Observability:
    
  - [ ] Tracing / [metrics](https://github.com/kamu-data/kamu-standards/blob/master/metrics_design.md): ✅
    
  - [ ] Alerts: ✅
- [ ] Documentation:
    
  - [ ] [Changelog](./CHANGELOG.md): ✅
    
  - [ ] [Public documentation](https://github.com/kamu-data/kamu-docs/): ✅
  
- [ ] Downstream effects:
    
  - [ ] [kamu-node](https://github.com/kamu-data/kamu-node): ✅
    
  - [ ] [kamu-web-ui](https://github.com/kamu-data/kamu-web-ui): ✅
    
  - [ ] [release-queue](https://github.com/orgs/kamu-data/projects/4/views/1)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,6 +12,8 @@ SQLITE_CRATES := ./src/infra/accounts/sqlite ./src/infra/auth-rebac/sqlite ./src
 ALL_DATABASE_CRATES := $(POSTGRES_CRATES) $(MYSQL_CRATES) $(SQLITE_CRATES)
 MIGRATION_DIRS := ./migrations/mysql ./migrations/postgres ./migrations/sqlite
 
+KAMU_CONTAINER_RUNTIME_TYPE ?= docker
```
Comment (sergiimk) [inline Makefile:15]: Should we default this to `podman` to match what our `ContainerRuntime` defaults to?

Not sure why we are running most unit tests with `podman` but all the DB setup.

Suggest poking #engineering channel to get peoples' opinion.

Comment (zaychenko-sergei) [reply Makefile:15]: Check with Mac lovers. It does not matter much for Linux env.

**Review by zaychenko-sergei — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (s373r) [reply Makefile:15]: > It does not matter much for Linux env.

On my Fedora the difference between `podman` and `docker` is very significant, I try not to use docker at all

Comment (s373r) [reply Makefile:15]: As an alternative option, use the default `docker` for macOS only

**Review by s373r — CHANGES_REQUESTED** [review]
Added a suggestion for implementation to the thread (https://github.com/kamu-data/kamu-cli/pull/991#discussion_r1886556294)

**Pull request merged**